crates/meroctl/src/cli/context/create.rs

```diff
@@ -73,56 +83,137 @@ async fn create_context(
             response.json().await?;
         let context = context_response.data.context;
 
-        println!("Context created successfully:");
-        println!("Context ID: {}", context.id);
-    } else {
-        let status = response.status();
-        let error_text = response.text().await?;
-        eyre::bail!(
-            "Request failed with status: {}. Error: {}",
-            status,
-            error_text
+        println!("Context `\x1b[36m{}\x1b[0m` created!", context.id);
+
+        println!(
+            "Context{{\x1b[36m{}\x1b[0m}} -> Application{{\x1b[36m{}\x1b[0m}}",
+            context.id, context.application_id
         );
+
+        return Ok(context.id);
     }
+
+    let status = response.status();
+    let error_text = response.text().await?;
+
+    eyre::bail!(
+        "Request failed with status: {}. Error: {}",
+        status,
+        error_text
+    );
+}
+
+async fn watch_app_and_update_context(
+    base_multiaddr: &Multiaddr,
+    context_id: calimero_primitives::context::ContextId,
+    path: Utf8PathBuf,
+    client: &Client,
+) -> eyre::Result<()> {
+    let (tx, mut rx) = mpsc::channel(1);
+
+    let handle = tokio::runtime::Handle::current();
+    let mut watcher = notify::recommended_watcher(move |evt| {
+        handle.block_on(async {
+            let _ = tx.send(evt).await;
+        })
+    })?;
```

Comment:
This is a blocking operation... I am concerned about calling a sync blocking operation in an async context to block on an async operation. Could we consider using `spawn_blocking()` here instead? That would use Tokio's dedicated blocking pool.

Reply:
`notify` [maintains it's own OS thread](https://github.com/notify-rs/notify/blob/128bf6230c03d39dbb7f301ff7b20e594e34c3a2/notify/src/fsevent.rs#L435-L437), which calls the closure once an event occurs, and that's why we're using `block_on`, and smuggling the tokio runtime handler across thread boundaries.

We shouldn't care about blocking in the closure because it executes in a non-async world, sending events to the async world with our `mpsc` channel which is non-blocking for tokio.